packages/klighd-core/src/bookmarks/bookmark-panel.tsx

```diff
@@ -0,0 +1,71 @@
+/** @jsx html */
+import { html } from "snabbdom-jsx"; // eslint-disable-line @typescript-eslint/no-unused-vars
+import { inject, injectable, postConstruct } from "inversify";
+import { VNode } from "snabbdom/vnode";
+import { DISymbol } from "../di.symbols";
+import { RenderOptionsRegistry } from "../options/render-options-registry";
+import { PreferencesRegistry } from "../preferences-registry";
+import { SidebarPanel } from "../sidebar";
+import { SynthesesRegistry } from "../syntheses/syntheses-registry";
+import { TYPES, IActionDispatcher, Action } from "sprotty";
+
+@injectable()
+export class BookmarkPanel extends SidebarPanel {
+    
+    @inject(TYPES.IActionDispatcher) private actionDispatcher: IActionDispatcher;
+    @inject(DISymbol.SynthesesRegistry) private synthesesRegistry: SynthesesRegistry;
+    @inject(DISymbol.PreferencesRegistry) private preferencesRegistry: PreferencesRegistry;
+    @inject(DISymbol.RenderOptionsRegistry) private renderOptionsRegistry: RenderOptionsRegistry;
+
+    private static bookmarkActions: [key: string, icon: VNode, action: Action][]
+
+    @postConstruct()
+    init(): void {
+        // Subscribe to different registry changes to make this panel reactive
+        this.synthesesRegistry.onChange(() => this.update());
+        this.preferencesRegistry.onChange(() => this.update());
+        this.renderOptionsRegistry.onChange(() => this.update());
```

Comment:
Should this panel really synchronize with these registries? Looks like they are not relevant for rendering its UI.

Reply:
added a BookmarkRegistry and only listen for changes there now